pennylane/control_flow/for_loop.py

```diff
@@ -114,6 +130,81 @@ def loop_rx(i, x):
         as well as the :doc:`sharp bits and debugging tips <catalyst:dev/sharp_bits>`
         page for an overview of using quantum just-in-time compilation.
 
+    .. details::
+        :title: Usage Details
+
+        A dynamically shaped array is an array whose shape depends on an abstract value. This is
+        an experimental jax mode that can be turned on with:
+
+        >>> import jax
+        >>> jax.config.update("jax_dynamic_shapes", True)
+
+        ``allow_array_resizing="auto"`` will try and choose between the following two possible modes.
+        If the needed mode is ``allow_array_resizing=False``, then this will require re-capturing
+        the loop, potentially taking more time.
+
+        When working with dynamic shapes in a ``for_loop``, we have two possible
+        options. ``allow_array_resizing=True`` treats every dynamic dimension as independent.
+
+        .. code-block:: python
+
+            @qml.for_loop(3, allow_array_resizing=True)
+            def f(i, x, y):
+                return jax.numpy.hstack([x, y]), 2*y
```

Comment:
```suggestion
                return jnp.hstack([x, y]), 2*y
```

I suggest using always `jnp` or `jax.numpy.array`

Reply:
I usually prefer the full `jax.numpy`, since it's one less import to remember.